### DISCUSSION TITLE
Removing the deprecation warnings

### DISCUSSION THREAD

**Discussion by lachesis** [nodesource/distributions#1678] [General]
If you want to run the old scripts (e.g. to install Node 14 on Ubuntu 22.04, which has no corresponding deb package in the new process), you can disable the deprecation warnings that add ~100 seconds with the following ugly hack:

```bash
curl -sL https://deb.nodesource.com/setup_14.x | grep -v '^[a-z]*_deprecation_warning$' | bash -
```

Our docker command is now:
```Dockerfile
RUN curl -sL https://deb.nodesource.com/setup_14.x | grep -v '^[a-z]*_deprecation_warning$' | bash - \
  && apt-get update && apt-get install -y nodejs \
  && rm -rf /var/lib/apt/lists/* \
  && apt-get clean
```

**Comment by cikaldev** [comment]
Today i just follow the suggest script, to skip deprecation warning. but my logs file become un-readable. And i just modify your suggestion script into more simple like this.

```shell
curl -fsSL https://deb.nodesource.com/setup_14.x | grep -v 'sleep' | bash -
```

The main problem is waiting 60+20+20 = 100s, before the script running. I think just skip the `sleep xx` and left the rest do their job. :laughing:

**Comment by riosje** [comment]
Guys what stop you from updating to the latests versions?
`Node18` and `Node20`

**Reply by cikaldev** [reply]
The company where i work, still use Legacy System that required v14 to run. If we talk about _what stop (me)_, that was the answer. But people never care there is "[Cobol Language](https://en.wikipedia.org/wiki/COBOL)" still exists from 1960, and used around the world. :laughing: 

Seriously if you remove the old version installer, people who need the specific version, will take the source and dirty build from source, for fulfilled their need on their business.

**Reply by riosje** [reply]
@cikaldev this is completely understandable that's the reason why we not remove those scripts.
In the other hand we have those waits and warnings because we want the people know they're using deprecated versions of Nodejs which is risky due to unpatched vulns.
I will think how to improve those warning and will decrease the sleep time.
